clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/VenicePushJob.java

```diff
@@ -1370,13 +1373,44 @@ protected static boolean shouldBuildCompressionDictionary(
     }
 
     LOGGER.info(
-        "No Compression dictionary will be generated with the compression strategy {} "
-            + "and compressionMetricCollectionEnabled is {}",
-        storeSetting.compressionStrategy,
-        (pushJobSetting.compressionMetricCollectionEnabled ? "Enabled" : "Disabled"));
+        "No Compression dictionary will be generated with the compression strategy"
+            + " {} and compressionMetricCollectionEnabled is disabled",
+        storeSetting.compressionStrategy);
     return false;
   }
 
+  /**
+   * This functions reevaluates {@link PushJobSetting#compressionMetricCollectionEnabled}
+   * based on the input data
```

Comment:
Can you add some more details here about why reevaluation is necessary?

Reply:
Added.